dlt/common/configuration/specs/azure_credentials.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional, Dict, Any
+
+from dlt.common import pendulum
+from dlt.common.exceptions import MissingDependencyException
+from dlt.common.typing import TSecretStrValue
+from dlt.common.configuration.specs import CredentialsConfiguration, CredentialsWithDefault, configspec
+from dlt.common.configuration.specs.exceptions import InvalidBoto3Session
+from dlt import version
+
+import fsspec
+
+
+@configspec
+class AzureCredentialsWithoutDefaults(CredentialsConfiguration):
+    """Credentials for azure blob storage, compatible with adlfs"""
+
+    azure_storage_account_name: str = None
+    azure_storage_account_key: Optional[TSecretStrValue] = None
+    azure_storage_sas_token: TSecretStrValue = None
+
+    def to_adlfs_credentials(self) -> Dict[str, Any]:
+        """Return a dict that can be passed as kwargs to adlfs"""
+        return dict(
```

Comment:
`adlfs` does not need this token. (maybe it does that itself?)
```
fs = fsspec.filesystem("adl", account_name=adls_account_name, account_key=adls_account_key)
print(fs.ls("dlt-ci-test-bucket/"))
```
works for me

Reply:
So the SAS is needed for snowflake when not using named stage https://github.com/dlt-hub/dlt/pull/592/files#diff-577441894b0e65b848310ea282c52bab00b6f649eb2c1b9402a5423d70e4d3e4R84    
Otherwise we don't need it